File: 2 período/aeds2/beecrowd/q2693.java
// https://judge.beecrowd.com/pt/problems/view/2693

import java.util.*;

class Pessoa {
  String nome;
  char regiao;
  int distancia;

  Pessoa(String nome, char regiao, int distancia) {
    this.nome = nome;
    this.regiao = regiao;
    this.distancia = distancia;
  }
}

public class q2693 {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);

    int n = sc.nextInt();
    sc.nextLine();
    Pessoa[] arr = new Pessoa[n];

    for (int i = 0; i < n; i++) {
      String tot = sc.nextLine();
      String[] separados = tot.split(" ");
      int value = Integer.parseInt(separados[2]);
      char c = separados[1].charAt(0);
      arr[i] = new Pessoa(separados[0], c, value);          
    }

    ordenaVet(arr);

    for (int i = 0; i < n; i++) {
      System.out.println(arr[i].nome);
    }
    sc.close();
  }

  public static void ordenaVet(Pessoa vet[]) {
    for (int i = 1; i < vet.length; i++) {
      Pessoa key = vet[i];
      int actual = i;
      while (actual > 0 && (vet[actual - 1].distancia > key.distancia || 
      (vet[actual - 1].distancia == key.distancia && vet[actual - 1].regiao > key.regiao) || (vet[actual - 1].distancia == key.distancia && vet[actual - 1].regiao == key.regiao && vet[actual - 1].nome.compareTo(key.nome) > 0)
      )) {
        vet[actual] = vet[actual - 1];
        actual--;
      }
      vet[actual] = key;
    }
  }
}

File: 2 período/aeds2/beecrowd/q2693.c
// https://judge.beecrowd.com/pt/problems/view/2693

#include <stdio.h>
#include <string.h>

struct pessoa {
    char nome[100];
    char regiao;
    int distancia;
};


void ordenaVet(struct pessoa* vet, int n) {
  for (int i = 1; i < n; i ++) {
    struct pessoa key = vet[i];
    int actual = i;

    while ((actual > 0) && (vet[actual - 1].distancia > key.distancia) ||
    (vet[actual - 1].distancia == key.distancia && vet[actual - 1].regiao > key.regiao) || (vet[actual - 1].distancia == key.distancia && vet[actual - 1].regiao == key.regiao && vet[actual - 1].nome < key.nome)
    ) {
      vet[actual] = vet[actual - 1];
      actual--;
    }

    vet[actual] = key;
  }
}


int main() {
  int n; scanf("%i", &n);
  struct pessoa arr[n];

  for (int i = 0; i < n; i++) {
        scanf("%s %c %d", arr[i].nome, &arr[i].regiao, &arr[i].distancia);
  }

  ordenaVet(arr, n);

  for (int i = 0; i < n; i ++) {
    printf("%s\n", arr[i].nome);
  }

  return 0;
}